File: Easy/Prime Factors/prime-factors.py
#User function Template for python3

class Solution:
	def AllPrimeFactors(self, N):
		l=[]
		for i in range(2,N+1):
		    while N%i==0:
		        if len(l)==0:
		            l.append(i)
		        elif len(l)!=0 and l[-1]!=i:
		            l.append(i)
		        N=N/i
		    if N==1:
		        break
		return l


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
	T=int(input())
	for i in range(T):
		N = int(input())
		ob = Solution()
		ans = ob.AllPrimeFactors(N)
		for _ in ans:
			print(_, end = " ")
		print()
# } Driver Code Ends

File: Difficulty: Easy/Prime Factors/prime-factors.java
//{ Driver Code Starts


import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
{
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine().trim());
        while(T-->0)
        {
            int N = Integer.parseInt(br.readLine().trim());
            Solution ob = new Solution();
            int[] ans = ob.AllPrimeFactors(N);
            for(int i = 0; i < ans.length; i++)
                System.out.print(ans[i] + " ");
            System.out.println();
        }
    }
}

// } Driver Code Ends




class Solution
{
    public int[] AllPrimeFactors(int N)
    {
        ArrayList<Integer> ans=new ArrayList<>();
        
        for(int i=2;i<=N;i++){
            if(N%i==0){
                ans.add(i);
                while(N%i==0){
                    N=N/i;
                }
            }
        }
        int limit=ans.size();
        int []finalAnswer=new int[limit];
        for(int i=0;i<limit;i++){
            finalAnswer[i]=ans.get(i);
        }
        return finalAnswer;
    }
}